document binary quantization (#281)
qdrant/landing_page: qdrant-landing/content/documentation/guides/quantization.md

Please refer to the [Quantization Tips](#quantization-tips) section for more information on how to optimize the quantization parameters for your use case.

## Binary Quantization

*Available as of v1.5.0*

> An extreme case of scalar quantization is binary quantization.

Reviewer: ```suggestion
Binary quantization is an extreme case of scalar quantization.
```

> It allows to represent each vector component as a single bit, effectively reducing the memory footprint by a **factor of 32**.

Reviewer: ```suggestion
This feature lets you represent each vector component as a single bit, effectively reducing the memory footprint by a **factor of 32**.
```

> At the same time, binary quantization is the fastest quantization method, as it allows to perform vector comparison using just a few CPU instructions.

Reviewer: ```suggestion
This is the fastest quantization method, since it lets you perform a vector comparison with a few CPU instructions.
```

> It can achieve up to **40x** speedup compared to the original vectors.

Reviewer: ```suggestion
Binary quantization can achieve up to a **40x** speedup compared to the original vectors.
```

However, binary quantization is only efficient for high-dimensional vectors and require a centered distribution of vector components. 

At the moment, binary quantization shows good accuracy results with the following models:

- OpenAI `text-embedding-ada-002` - 1536d tested with [dbpedia dataset](https://huggingface.co/datasets/KShivendu/dbpedia-entities-openai-1M) achieving 0.98 recall@100 with 4x oversampling
- Cohere AI `embed-english-v2.0` - 4096d tested on [wikipedia embeddings](https://huggingface.co/datasets/nreimers/wikipedia-22-12-large/tree/main) - 0.98 recall@50 with 2x oversampling

Models with a lower dimensionality or a different distribution of vector components may require additional experiments to find the optimal quantization parameters.

> It is recommended to use binary quantization only with rescoring enabled, as it can significantly improve the search quality

Reviewer: ```suggestion
We recommend using binary quantization only with rescoring enabled, as it can significantly improve the search quality
```

with just a minor performance impact.
Additionally, oversampling can be used to tune the tradeoff between search speed and search quality in the query time.

### Binary Quantization as Hamming Distance

> Usage of binary quantization have an additional benefit - it allows to efficiently emulate Hamming distance with dot product.

Reviewer: ```suggestion
The additional benefit of this method is that lets you efficiently emulate Hamming distance with dot product.
```

> More specifilly, if original vectors contain `{-1, 1}` as possible values, then dot product of two vectors is equal to Hamming distance by simply replacing `-1` with `0` and `1` with `1`.

Reviewer: ```suggestion
Specifically, if original vectors contain `{-1, 1}` as possible values, then the dot product of two vectors is equal to the Hamming distance by simply replacing `-1` with `0` and `1` with `1`.
```

<!-- hidden section -->

<details>

>   <summary><b>Example of the truth table</b></summary>

Reviewer: ```suggestion
  <summary><b>Sample truth table</b></summary>
```
